packages/components/src/ListView/Items/Item/Item.component.js

```diff
@@ -34,12 +40,23 @@ function Item(props) {
 							id={itemId}
 							type="checkbox"
 							checked={item.checked}
-							onChange={event => item.onChange(event, item)}
+							onChange={event =>
+								parentItem ? item.onChange(event, item, parentItem) : item.onChange(event, item)
```

Comment:
Looks like `parentItem` could be passed in every cases and its value tested inside the `onChange` function.